addons/sale/security/sale_security.xml

```diff
@@ -91,7 +91,7 @@
     <record id="sale_order_personal_rule" model="ir.rule">
         <field name="name">Personal Orders</field>
         <field ref="model_sale_order" name="model_id"/>
-        <field name="domain_force">['|',('user_id','=',user.id),('user_id','=',False)]</field>
+        <field name="domain_force">['|','|', ('create_uid','=',user.id), ('user_id','=',user.id), ('user_id','=',False)]</field>
```

Comment:
Good idea. I will be always able to create sale orders independently from what the group limitates me.